File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.js
function solution(numLog) {
    let answer = '';
    let num = numLog[0]
    for(let i = 1; i < numLog.length; i++){
        if(numLog[i] == num + 1){
            answer += 'w';
            num++;
        }
        else if(numLog[i] == num - 1){
            answer += 's';
            num--;
        }
        else if(numLog[i] == num + 10){
            answer += 'd';
            num += 10;
        }
        else {
            answer += 'a';
            num -= 10;
        }
    }
    return answer;
}

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        StringBuilder sb = new StringBuilder();
        int num = numLog[0];
        for(int i = 1; i < numLog.length; i++){
            if(numLog[i] == num + 1){
                sb.append("w");
                num++;
            }
            else if(numLog[i] == num - 1){
                sb.append("s");
                num--;
            }
            else if(numLog[i] == num + 10){
                sb.append("d");
                num += 10;
            }
            else{
                sb.append("a");
                num -= 10;
            }
                
        }
        String result = sb.toString();
        return result;
    }
}